drgn/helpers/linux/module.py

```diff
@@ -0,0 +1,228 @@
+# Copyright (c) 2024 Oracle and/or its affiliates
+# SPDX-License-Identifier: LGPL-2.1-or-later
+"""
+Modules
+-------
+
+The ``drgn.helpers.linux.module`` module contains helpers for working with
+loaded kernel modules.
+"""
+import operator
+from typing import Iterable, List, Tuple, Union
+
+from drgn import NULL, IntegerLike, Object, Program
+from drgn.helpers.common.prog import takes_program_or_default
+from drgn.helpers.linux.list import list_for_each_entry
+from drgn.helpers.linux.rbtree import rb_find
+
+__all__ = (
+    "address_to_module",
+    "for_each_module",
+    "lookup_module",
+    "module_address_regions",
+    "module_percpu_region",
+)
+
+
+def for_each_module(prog: Program) -> Iterable[Object]:
+    """
+    Returns all loaded kernel modules
+
+    :returns: Iterable of ``struct module *`` objects
+    """
+    return list_for_each_entry("struct module", prog["modules"].address_of_(), "list")
+
+
+@takes_program_or_default
+def lookup_module(prog: Program, name: Union[str, bytes]) -> Object:
```

Comment:
The equivalent kernel function is named [`find_module()`](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/linux/module.h?h=v6.10-rc7#n657), so let's use that name.